src/App/Formatter/Post/CSV.php

```diff
@@ -228,61 +228,73 @@ private function getValueFromRecord($record, $keyParam, $attributes)
     {
         // assume it's empty since we go through this for all attributes which might not be available
         $return = '';
+        $should_return = false;
         // the $keyParam is the key=>label we get in createSortedHeading (keyLabel.index)
         $keySet = explode('.', $keyParam); //contains key + index of the key
         $headingKey = isset($keySet[0]) ? $keySet[0] : null; // the heading type (sets, contact, title)
         $key = isset($keySet[1]) ? $keySet[1] : null; // the key to use (0, lat,lon)
         // check that the key we received is available in $attributes
         $recordAttributes = isset($attributes[$headingKey]) ? $attributes[$headingKey] : null;
-        // Ignore attributes that are not related to this Post by Form Id
-        // Ensure that native attributes identified via id 0 are included
-        if (is_array($recordAttributes)
-            && isset($recordAttributes['form_id'])
-            && isset($record['form_id'])
-            && $recordAttributes['form_id'] != 0
-            && ($record['form_id'] != $recordAttributes['form_id'])) {
-            return '';
-        }
+
         // If the returned attribute for the given heading key is the native form name attribute
         // Retrieve Form Name from the attribute rather than from the Post until the data model improves
-        if (is_array($recordAttributes)
-            && isset($recordAttributes['type'])
+        
+        if (is_array($recordAttributes) && isset($recordAttributes['type'])
             && $recordAttributes['type'] === 'form_name') {
-            return is_array($recordAttributes) && isset($recordAttributes['form_name'])
-                ? $recordAttributes['form_name'] : '';
+            return is_array($record) && isset($record['form_name']) ? $record['form_name'] : 'Unstructured';
+        }
+        // Ignore attributes that are not related to this Post by Form Id
+        // Ensure that native attributes identified via id 0 are included
+        if (is_array($recordAttributes) && isset($recordAttributes['form_id']) && isset($record['form_id'])
+            && $recordAttributes['form_id'] != 0 && ($record['form_id'] != $recordAttributes['form_id'])) {
+            $should_return = true;
+        }
+
+        // Check if we are dealing with a structured post but not a structured attribute
+        if (is_array($recordAttributes) && isset($recordAttributes['unstructured'])
+            && $recordAttributes['unstructured'] && isset($record['form_id'])) {
+            $should_return = true;
         }
+
+        // Check if we're dealing with an unstructured post but a structured attribute
+        if (!isset($record['form_id'])
+            && isset($recordAttributes['form_id']) && $recordAttributes['form_id'] != 0) {
```

Comment:
wait the attributes can be valid and have a form id but not be in a record with a form id? I missed that yesterday :(